lib/modules.nix

```diff
@@ -87,6 +84,16 @@ rec {
       result = { inherit options config; };
     in result;
 
+
+ /* Filter disabled modules. Modules can be disabled allowing
+    their implementation to be replaced. */
+ filterModules = modules:
+   let
+     moduleKey = m: if isString m then toString ../nixos/modules + "/" + m else toString m;
```

Comment:
`../nixos/modules` suppose that all modules that would be replaced would be in NixOS repository.
I honestly does not like that because this gives option to NixOS which are not available to others.

This does not support the idea that someone else might want to use the module system for something else, such as NixUP, or extend NixOS with an additional set of default modules.